frontend/src/app/search/SearchForm.tsx

```diff
@@ -13,21 +16,23 @@ interface SearchResultsListProps {
 const SearchResultsList: React.FC<SearchResultsListProps> = ({
   searchResults,
 }) => {
-  // useFormStatus only works here because this component's
-  // parent is the form
   const { pending } = useFormStatus();
```

Comment:
Should a state instead be passed so that this component is not dependent on a parent function?

Reply:
Next and React are pushing this new pattern with `useFormState` and `[useFormStatus](https://react.dev/reference/react-dom/hooks/useFormStatus)` . I think the `SearchResultsList` component should be a child of the SearchForm - I just wanted to note this (their docs give several warnings about it also)

( we get access to other things as well `const { pending, data, method, action } = useFormStatus()`)

Reply:
Noting that NextJS recommends using the URL and search params to manage state with search pages: https://nextjs.org/learn/dashboard-app/adding-search-and-pagination#why-use-url-search-params

Is there a reason not to use that paradigm?

Reply:
Y'all know better than me. I just thought components were supposed to be self reliant, and not depend on specific parents. Not that we'd ever use this component in another context; I was mostly thinking about showing it in isolation in Storybook.

Reply:
I'm also confused about this component being named "SearchForm" but it's not a form, its parent is the form.

Reply:
>I'm also confused about this component being named "SearchForm" but it's not a form, its parent is the form.

@andycochran - that file does have the `form` tag but it's not your traditional form (registration form etc..) . It could be called `SearchLayout` or something else if that makes more sense?